Milestones/Milestone9 Meme Generator/Milestone9 Meme Generator/ViewController.swift

```diff
@@ -0,0 +1,199 @@
+//
+//  ViewController.swift
+//  Milestone9 Meme Generator
+//
+//  Created by Jason Pinlac on 3/18/20.
+//  Copyright © 2020 Jason Pinlac. All rights reserved.
+//
+
+import UIKit
+
+class ViewController: UIViewController, UITextFieldDelegate, UIImagePickerControllerDelegate, UINavigationControllerDelegate {
```

Comment:
In general, you will want to make extensions to ViewController for protocol/delegate conformance. 

So for example, 

```swift 

class ViewController: UIViewController {
...
}

extension ViewController: UITextFieldDelegate {
    func textFieldShouldReturn(_ textField: UITextField) -> Bool {
         textField.resignFirstResponder()
         return true
     }
}

extension ViewController: UIImagePickerControllerDelegate {
    func imagePickerController(_ picker: UIImagePickerController, didFinishPickingMediaWithInfo info:    [UIImagePickerController.InfoKey : Any]) {
         guard let image = info[UIImagePickerController.InfoKey.editedImage] as? UIImage else { return }
         selectedImage = image
         picker.dismiss(animated: true)
         renderImage()
         topTextField.isEnabled = true
         bottomTextField.isEnabled = true
         topTextSizeSlider.isEnabled = true
         bottomTextSizeSlider.isEnabled = true
     }
}

```
2 things happen when you do this, 

Xcode delineates in the jump bar so you can find things faster and it organizes your code.

Reply:
Okay cool. Just to be clear. Here I'm extending the ViewController class by  adopting a protocol and implementing the textFieldShouldReturn(..) method and the same goes for the UIImagePickerControllerDelegate and its method. I do this because using the jump bars CTRL-6 will allow the methods to be organized and faster to use? Also, you separated the protocols as 2 extensions. Is it okay to use a single extension block and include all conforming protocols or is it better to separate them within their own extension block like shown above. Thanks! Also, you're right about the code being much more organized. Now apple's protocol conforming methods aren't mixed in with my class methods.

Reply:
You will want an extension for each delegate protocol rather than one. 

> Now apple's protocol conforming methods aren't mixed in with my class methods.

Thats the idea